include/mercury/kw/kw.h

```diff
@@ -34,25 +35,38 @@ class KanoWorld
      */
     explicit KanoWorld(std::shared_ptr<IHTTPClient> client =
         std::make_shared<HTTPClient>());
-    bool login(string username, string password, bool verbose);
-    bool refresh_token(string token, bool verbose);
+
+    bool login(const string& username, const string& password,
+               const bool verbose = false);
+    bool logout(const bool verbose = false);
+    bool refresh_token(string token, const bool verbose = false);
 
     string get_hostname(string config_filename);
     string get_refresh_header(string token);
 
-    static size_t write_function(void *ptr, size_t size, size_t nmemb, void *stream);
+    bool is_logged_in(const bool verbose = false);
+    string whoami();
 
-    bool am_i_logged_in(void);
-    string whoami(void);
 
     /**
      * \brief Determine if the user's account has been verified with parental
      *        permission.
      */
     bool is_account_verified();
 
+    const string data_filename;
+    string get_token() const;
+    string get_expiration_date() const;
+    bool load_data();
+
  private:
     std::shared_ptr<IHTTPClient> http_client;
+    std::string parse_token(const std::shared_ptr<JSON_Value> res) const;
```

Comment:
You have a `using std::string` defined at the top, but in a few places here you use `std::string` and `string`. Recommend picking one and keeping it consistent.

If you go with `using`, then you could add one for `using std::shared_ptr;` as well.